File: 1342-Number of Steps to Reduce a Number to Zero/C++/iterative.cpp
class Solution {
public:
    int numberOfSteps(int num) {
        int step = 0;
        while(num!=0){
            step += 1;
            if(num%2 == 0){
                num = num / 2;
            }else{
                num = num -1;
            }
        }
        return step;
    }
};

File: 1342-Number of Steps to Reduce a Number to Zero/Python/iterative.py
class Solution:
    def numberOfSteps(self, num: int) -> int:
        steps = 0
        evenOrOdd = lambda num : (num//2) if (num % 2 == 0) else (num-1)
        add = lambda steps : steps+1
        while(num!=0):
            num = evenOrOdd(num)
            steps = add(steps)
        return steps